.github/workflows/ci.yml

```diff
@@ -4,67 +4,72 @@ on:
   push:
     branches:
       - main
+  workflow_call:
+
 env:
   python_version: '3.9'
 
+defaults:
+  run:
+    shell: bash  # https://github.com/beeware/briefcase/pull/912
+
 jobs:
   pre-commit:
     name: Pre-commit code style checks
     runs-on: ubuntu-latest
     timeout-minutes: 10
     steps:
       - uses: actions/checkout@v3.1.0
-        with:
-          fetch-depth: 0
-          ref: ${{ github.event.pull_request.head.sha }}
       - name: Set up Python
         uses: actions/setup-python@v4.2.0
         with:
           python-version: ${{ env.python_version }}
       - name: Lint with Pre-commit
         uses: pre-commit/action@v3.0.0
 
-  beefore:
-    name: Pre-test checks
-    needs: pre-commit
+  towncrier:
     runs-on: ubuntu-latest
-    strategy:
-      max-parallel: 4
-      matrix:
-        task:
-        - 'towncrier-check'
-        - 'package'
     steps:
-    # Check out main; needed for towncrier comparisons.
+    # Fetch main branch for comparison, then check out current branch.
     - uses: actions/checkout@v3.1.0
       with:
         fetch-depth: 0
         ref: main
     - uses: actions/checkout@v3.1.0
       with:
         fetch-depth: 0
-    - name: Set up Python
-      uses: actions/setup-python@v4.2.0
+    - uses: actions/setup-python@v4.2.0
       with:
         python-version: ${{ env.python_version }}
-    - name: Install dependencies
-      run: |
-        pip install --upgrade pip
-        pip install --upgrade setuptools
-        pip install setuptools_scm tox
-    - name: Run pre-test check
-      run: |
-        tox -e ${{ matrix.task }}
+    - run: pip install tox
+    - run: tox -e towncrier-check
+
+  package:
+    runs-on: ubuntu-latest
+    steps:
+    # Fetch all refs so setuptools_scm can generate the correct version number.
+    - uses: actions/checkout@v3.1.0
+      with:
+        fetch-depth: 0
+    - uses: actions/setup-python@v4.2.0
+      with:
+        python-version: ${{ env.python_version }}
+    - run: pip install tox
+    - run: tox -e package
+    - uses: actions/upload-artifact@v3
+      with:
+        name: packages
+        path: dist
+        if-no-files-found: error
 
   unit-tests:
     name: Unit tests
-    needs: beefore
+    needs: [pre-commit, towncrier, package]
     strategy:
-      max-parallel: 4
```

Comment:
I'm not sure why this was here, but removing it allows all 12 jobs to run at the same time, saving at least 10 minutes over the whole CI run.